### PR TITLE
Timeline Footer Width

### PR DESCRIPTION
**Pull request by DilanRamirez**

### CODE REVIEW

File: src/components/user-view/document-timeline/timeline-footer.tsx [hunk 1]
```diff
@@ -81,6 +205,7 @@ export default function TimelineFooter(props: {
  exists, it calculates whether the content inside the element overflows horizontally. */
   useEffect(() => {
     if (footerTimelineRef.current === null) return;
+    dispatch(loadTimelinePoints({ timelinePoints }));
```
Comment (aaronshiel) [inline src/components/user-view/document-timeline/timeline-footer.tsx:208]: 2 things:

- we shouldn't ever call dispatch from within a functional component. Anything like this should be moved into a hook function (e.g. use-with-state.tsx) 
- it looks like you're adding timelinePoints into the redux state just to be able to check its length in the QontoStepIcon component, which is an over-implementation. You should be able to just pass down this info from one of its parent components that already has the timelinePoints data (looks like the TimelineFooter component has this data)

File: src/components/user-view/document-timeline/timeline-footer.tsx [hunk 2]
```diff
@@ -91,60 +216,79 @@ export default function TimelineFooter(props: {
...
+                    onMouseEnter={() => handleMouseEnter(i)}
+                    onMouseLeave={handleMouseLeave}
+                  >
+                    {hoverIndex !== i ? (
+                      <div className="timeline-footer-item-card-inner">
+                        <Typography
+                          className="text-2"
+                          style={{ textAlign: 'center' }}
+                        >
+                          {formatISODateToReadable(timelinePoint.versionTime) ||
```
Comment (aaronshiel) [inline src/components/user-view/document-timeline/timeline-footer.tsx:272]: remove `|| ''`

File: src/components/user-view/document-timeline/timeline-footer.tsx [hunk 3]
```diff
@@ -91,60 +216,79 @@ export default function TimelineFooter(props: {
...
+                            ''}
+                        </Typography>
+                        <Typography className="text-3">
+                          {convertDateTimelinePointTime(
+                            timelinePoint.versionTime
+                          ) || ''}
```
Comment (aaronshiel) [inline src/components/user-view/document-timeline/timeline-footer.tsx:278]: remove `|| ''`

File: src/helpers.ts [hunk 4]
```diff
@@ -252,3 +252,55 @@ export function hasMinutesPassed(
...
+    hour: 'numeric',
+    minute: 'numeric',
+    hour12: true,
+    timeZone: 'UTC',
+  };
+
+  return timestamp.toLocaleString('en-US', options);
+};
+
+/**
```
Comment (aaronshiel) [inline src/helpers.ts:282]: This might be a personal opinion, but these long paragraphs seem to be overexplaining the code

**Review by aaronshiel — CHANGES_REQUESTED** [review]

File: src/components/user-view/document-timeline/timeline-footer.tsx (hunk 1 above)
Comment (DilanRamirez) [reply src/components/user-view/document-timeline/timeline-footer.tsx:208]: It looks like it does not accept props from parent component. I opted for another solution much more easies only using icons.

File: src/store/slices/state/index.ts [hunk 5]
```diff
@@ -37,6 +38,7 @@ export interface State {
   overideGptModel: GptModels;
   viewingRole: UserRole;
   viewingAdvancedOptions: boolean;
+  timelinePoints: GQLTimelinePoint[];
```
Comment (aaronshiel) [inline src/store/slices/state/index.ts:41]: remove from redux if no longer used

**Review by aaronshiel — CHANGES_REQUESTED** [review]

File: src/store/slices/state/index.ts (hunk 5 above)
Comment (DilanRamirez) [reply src/store/slices/state/index.ts:41]: removed

Comment (DilanRamirez) [reply src/store/slices/state/index.ts]: removed

**Review by aaronshiel — APPROVED** [review]

**Pull request merged**